### PR TITLE
[FLOSS T2] JSON spelling

### LINKED ISSUE #4212: [FLOSS] JSON spelling
While finishing up #4190, I noticed that the spell checker script does not allow `JSON` to be spelled as `json` and the build failed.
This means that commands like
```
kdb export system:/tests/passwd/root json
```
get flagged as wrong spelling.
The easiest solution would be to allow `JSON` to be spelled as `json`. 
@markus2330 also pointed out that it would be also possible to have the spell checker ignore the _case_  for commands like `kdb mount/export`.

### PR DESCRIPTION
**Pull request by muskater**
## Basics

This fixes #4212 

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: scripts/sed [hunk 1]
```diff
@@ -65,7 +65,7 @@ s:\<Key/value:Key-value:g
 
 s/[^#"]\<data-?type/data type/g
 
-s/([^\./`-])[Jj]son/\1JSON/
+s/([^\./`-])[J]son/\1JSON/
```
Comment (markus2330) [inline scripts/sed:68]: Better to change from any spelling of jSoN to json; and any spelling of JsOn to JSON?

Comment (muskater) [reply scripts/sed:68]: I agree, while these types of spelling will probably be rare, it is better if we make sure to fix them.

Comment (markus2330) [reply scripts/sed:68]: You never know where a typo will happen :wink: 

Btw. character class with only one character is not needed.

**Comment by markus2330** [comment]
Thank you very much! :fireworks:

**Comment by Dynamichost96** [comment]
## Documentation is introductory, concise, good to read and describes everything what the PR does

Commit messages are good, and Documentation is excellent and understandable.

## Examples are well chosen and understandable
No new examples were provided

##  Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
No new code changes where made

##  APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
No new API where added

##  Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)
No new code changes where made

Review has been done together with @Bujuhu

**Pull request merged**